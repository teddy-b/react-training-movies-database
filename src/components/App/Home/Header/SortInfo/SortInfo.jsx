import React from 'react'

import { SORT_BY } from '../../../../../constants'
import './SortInfo.scss'

const SortInfo = ({ count, onSort }) => (
  <div className="sortInfo">
    {count > 0 && <div>{count} movies found</div>}
    {count > 0 && <div>
      <span>Sort by </span>
      <a onClick={() => onSort(SORT_BY.releaseDate)}>Release date</a> 
      <a onClick={() => onSort(SORT_BY.rating)}>Rating</a>
    </div>}
  </div>
)

export default SortInfo
