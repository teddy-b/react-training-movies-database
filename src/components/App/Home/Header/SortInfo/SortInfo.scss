@import '../../../../../styles/constants.scss';

.sortInfo {
  display: flex;
  justify-content: space-between;
  background-color: $lightgrey;
  padding: .5em;

  a {
    margin: .5em;
    cursor: pointer;
  }
}
