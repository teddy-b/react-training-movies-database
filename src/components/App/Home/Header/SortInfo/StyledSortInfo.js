/* @flow */

import styled from 'styled-components'

import { black, grey1, red } from '../../../../../constants/styles'

export const StyledSortInfo = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: ${grey1};
  padding: .5em;
`

export const StyledButton = styled.button`
  border: none;
  background-color: inherit;
  font-size: inherit;
  cursor: pointer;
  color: ${(props: { [prop: string]: string }): string => (props.selected ? red : black)};
`
