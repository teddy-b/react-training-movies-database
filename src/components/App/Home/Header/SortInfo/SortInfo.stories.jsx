/* @flow */

import * as React from 'react'

import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import type { Renderable } from '@storybook/react'

import SortInfo from './SortInfo'
import { SORT_BY } from '../../../../../constants/global'

const props = {
  count: 24,
  onSortMoviesByRating: action('clicked'),
  onSortMoviesByRelaseDate: action('clicked'),
  sortBy: SORT_BY.releaseDate
}

storiesOf('Components/SortInfo', module)
  .add('sort by release date', (): Renderable => (
    <SortInfo
      {...props}
      sortBy={SORT_BY.releaseDate}
    />
  ))
  .add('sort by rating', (): Renderable => (
    <SortInfo
      {...props}
      sortBy={SORT_BY.rating}
    />
  ))
