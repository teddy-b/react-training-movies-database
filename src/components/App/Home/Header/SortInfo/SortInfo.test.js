import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SortInfo from './SortInfo'
import { moviesMock } from '../../../../../mocks/movies-mocks'

describe('SortInfo', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <SortInfo
        count={moviesMock.length}
        onSort={jest.fn()}
      />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
  
  it('should trigger onSort onClick', () => {
    const onSortMock = jest.fn()
    const component = shallow(<SortInfo
      count={moviesMock.length}
      onSort={onSortMock}
    />)
  
    component.find('a').at(0).simulate('click')
  
    expect(onSortMock.mock.calls.length).toBe(1)
  
    component.find('a').at(1).simulate('click')
  
    expect(onSortMock.mock.calls.length).toBe(2)
  })
})
