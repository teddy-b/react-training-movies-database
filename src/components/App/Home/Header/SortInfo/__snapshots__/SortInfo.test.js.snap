// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortInfo renders correctly 1`] = `
<div
  className="sortInfo"
>
  <div>
    10
     movies found
  </div>
  <div>
    <span>
      Sort by 
    </span>
    <a
      onClick={[Function]}
    >
      Release date
    </a>
    <a
      onClick={[Function]}
    >
      Rating
    </a>
  </div>
</div>
`;
